### PR TITLE
[cntt] Contributing.rst

### PR DESCRIPTION
**Pull request by pgoyal01**

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
+Contribution
+============
+
+This document is an extension to the generic `Anuket Project Operations and Procedures
+<https://wiki.anuket.io/display/HOME/Anuket+Project+Operations+and+Procedures>
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:5]: @pgoyal01 The ending characters are missing for this external links (`__)

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+Permission levels
+-----------------
+
+There are different permissions (levels) available for contributing into the Anuket specifications:
+- **Outside Collaborators**: This includes general public.
+  - Outside collaborators can create issues and add comments to issues and Pull Requests.
+- **Read, Triage, Write, and Maintain** Permissions, which are explained in
+  `here <https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/
+  repository-permission-levels-for-an-organization#permission-levels-for-repositories-owned-by-an-organization>,
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:18]: the same here (missing ending characters), am I wrong?

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+<https://wiki.anuket.io/display/HOME/Anuket+Project+Operations+and+Procedures>
+and describes contribution rules to the Anuket specifications, that are the set of
+documents maintained in this repository.
+
+
+Permission levels
+-----------------
+
+There are different permissions (levels) available for contributing into the Anuket specifications:
+- **Outside Collaborators**: This includes general public.
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:14]: @pgoyal01 A blank line is missing before the list

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  - Outside collaborators can create issues and add comments to issues and Pull Requests.
+- **Read, Triage, Write, and Maintain** Permissions, which are explained in
+  `here <https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/
+  repository-permission-levels-for-an-organization#permission-levels-for-repositories-owned-by-an-organization>,
+  will be assigned as follows:
+  - **Read** will be assigned to those who want to create issues, join the discussions, receive notifications,
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:20]: A blank line should be added before the second list

File: CONTRIBUTING.rst (hunk 1 above)
Comment (pgoyal01) [inline CONTRIBUTING.rst:5]:
```suggestion
<https://wiki.anuket.io/display/HOME/Anuket+Project+Operations+and+Procedures>`

```

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  has been reached on the PR. This triggers an action to resolve the PR by the TSC.
+- **Forced**: This label will be **automatically** applied to PRs to indicate that the PR has
+  been merged without following the default process (due to reasons identified below)
+
+
+Issues
+------
+
+- Title needs to be in the following format [SPx Chx] (if applicable).
+  - SPx: The name of the sub-project it is targeting.
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:52]: Blank line missing

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+  - Chx: The name of the chapter it is targeting (if applicable).
+- Upon Creation, Issues must be correctly set with the correct **Project Name**. 
+  - Project Name = Sub-project name.
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:55]: Blank line missing

File: CONTRIBUTING.rst (hunk 3 above)
Comment (pgoyal01) [inline CONTRIBUTING.rst:14]:
```suggestion

- **Outside Collaborators**: This includes general public.

```

File: CONTRIBUTING.rst (hunk 2 above)
Comment (pgoyal01) [inline CONTRIBUTING.rst:18]:
```suggestion
  repository-permission-levels-for-an-organization#permission-levels-for-repositories-owned-by-an-organization>`,

```

File: CONTRIBUTING.rst (hunk 4 above)
Comment (pgoyal01) [inline CONTRIBUTING.rst:20]:
```suggestion

    - **Read** will be assigned to those who want to create issues, join the discussions, receive notifications,

```

File: CONTRIBUTING.rst [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+- A PR for an issue must only be created by the contributor assigned to the issue (either by
+  self assignment or by the respective sub-project lead).
+- Only the person assigned ("assignee") to a PR may edit it. The normal process is for others
+  to make suggests in "Comments" but not directly edit a PR unless the assignee has granted
+  another contributor permission to contribute directly to the PR.
+- A PR is recommended to only create/modify content within the scope of a single chapter in a
+  single sub-project if possible.
+- A PR should only add/change content related to the issue associated with the PR.
+- Comments/sub-Conversations within a PR may only be marked Resolved by:
+  - A) The originator of the sub-conversation.
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:83]: Blank line missing

File: CONTRIBUTING.rst [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+  - **Do not modify an existing commit, create separate commits under the same PR**.
+  - After major changes to a PR reapproval is needed from all previous approvers
+  - PRs will be merged using "Squash and Merge" to enforce a linear commit history.
+
+
+Pull Requests Approval Process
+------------------------------
+
+- Once a PR is created, it needs to get the following approvals before it is merged into master.
+  - From at least three of contributors for that sub-project
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:98]: Blank line missing

File: CONTRIBUTING.rst [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+   Approval Process
+
+
+Idle Pull Requests Policy
+-------------------------
+
+Pull Requests will be automatically labled as "**Idle**" when:
+- No engagement/activity (content, reviews, conversations) from author, reviewers and sub-project
+  contributors on non-merge ready PRs for 15 calendar days.
+  - Personal holidays or public holidays will not be counted.
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:134]: Blank line missing

File: CONTRIBUTING.rst [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+  - Sub-project lead can request additional approvals reviews.
+    - Recommendation is to use 4 approvals for complex PRs.
+  - 2 business days waiting period should be applied before Final approval.
+- The selection of which contributor to approve a PR is made by the sub-project lead and
+  should take those factors into consideration:
+  - Contributor needs to be actively discussing the PR to be selected for approval.
+- PRs will be merged automatically online by the sub-project lead or other committers of the
+  sub-project once consensus is reached on the changes, all issues are resolved, and all of
+  the required approvals/re-approvals are received. 
+  - Sub-projects should have an agreed set of committers who are trusted with merging rights
```
Comment (karinesevilla) [inline CONTRIBUTING.rst:110]: Blank line missing

File: CONTRIBUTING.rst [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+    to the sub-projects. [CODEOWNERS](CODEOWNERS) file should contain all the committers of
```
Comment (pgoyal01) [inline CONTRIBUTING.rst:111]:
```suggestion
    to the sub-projects. `CODEOWNERS <https://github.com/cntt-n/CNTT/blob/master/CODEOWNERS>` file should contain all the committers of

```

**Review by pgoyal01 — COMMENTED** [review]
@karinesevilla Thanks

File: CONTRIBUTING.rst (hunk 5 above)
Comment (pgoyal01) [reply CONTRIBUTING.rst:58]: @karinesevilla For the sub-bullets a blank line is not required

File: CONTRIBUTING.rst [hunk 12]
```diff
@@ -0,0 +1,191 @@
+Contribution
+============
+
+This document is an extension to the generic `Anuket Project Operations and Procedures
+<https://wiki.anuket.io/display/HOME/Anuket+Project+Operations+and+Procedures>`__
+
```
Comment (pgoyal01) [inline CONTRIBUTING.rst:6]:
```suggestion

```

File: CONTRIBUTING.rst [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+Permission levels
+-----------------
+
+There are different permissions (levels) available for contributing into the Anuket specifications:
+
+- **Outside Collaborators**: This includes general public.
+
+  - Outside collaborators can create issues and add comments to issues and Pull Requests.
+- **Read, Triage, Write, and Maintain** Permissions, which are explained in
```
Comment (pgoyal01) [inline CONTRIBUTING.rst:18]:
```suggestion

- **Read, Triage, Write, and Maintain** Permissions, which are explained in

```

File: CONTRIBUTING.rst (hunk 4 above)
Comment (pgoyal01) [reply CONTRIBUTING.rst:24]: @karinesevilla Thanks - Fixed

File: CONTRIBUTING.rst (hunk 5 above)
Comment (pgoyal01) [reply CONTRIBUTING.rst:57]: @karinesevilla Thanks - Fixed

File: CONTRIBUTING.rst (hunk 6 above)
Comment (pgoyal01) [reply CONTRIBUTING.rst:62]: @karinesevilla Thanks - Fixed

File: CONTRIBUTING.rst [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+
+
+Pull Requests
+-------------
+
+- Minor editorial change PRs can be raised without a corresponding Issue, but must be labelled
+  as "Minor". Otherwise, PRs must only be created when there is an issue present and a decision
+  to create a PR is made. 
+- A PR must reference the issue it is resolving in the description field.
+   - using GitHub predefined keyword "Fixes Issue#" will automatically close the issue which
```
[the remark this answers is not in the document]
Comment (pgoyal01) [reply CONTRIBUTING.rst:83]: @karinesevilla Thanks - Fixed

File: CONTRIBUTING.rst (hunk 8 above)
Comment (pgoyal01) [reply CONTRIBUTING.rst:114]: @karinesevilla Thanks - Fixed

**Review by collivier — APPROVED** [review]

**Pull request merged**